hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+  },
+  foot: {
+    text: ["tspan", "All ticks are in units of &#xbd;T",
+           ["tspan", {'baseline-shift':'sub'}, "sck"],
+          "=&#xbd;T",
+           ["tspan", {'baseline-shift':'sub'}, "core"],
+          "&#xd7;(CLKDIV+1)"]
+  }
+}
+{{< /wavejson >}}
+
+These settings are all minimum bounds, and delays in the FSM implementation may create more margin in each of these timing constraints.
+
+### Special Command Fields
+
+The {{< regref "COMMAND" >}} register must be written every time a command is issued.
+Whenever a `1` is written to {{< regref "COMMAND.GO" >}}, contents of the {{< regref "CONFIG" >}} and {{< regref "COMMAND" >}} registers are passed through the Config/Command CDC to the SPI_HOST core FSM.
+Once the command has started, the core will assert {{< regref "STATUS.ACTIVE" >}}.
+The command is complete when {{< regref "STATUS.ACTIVE" >}} goes low.
+An `event` interrupt can also be triggered to go off on completion by setting {{< regref "EVENT_ENABLE.CMD_CMPLT" >}}.
+
+**TODO**: Describe TX1, TXN, dummy, RXN etc.  Refer to previous waveform.
+
+### Chip Select ID's
+
+Each instance of the SPI_HOST IP support up to 8 chip select lines (`cs_n[MaxCS-1:0]`).
+Each `cs_n` line can be routed either to a single peripheral or to a daisy-chain of peripherals.
+The three bit {{< regref "COMMAND.CSID" >}} field in each command indicates that the line `cs_n[COMMAND.CSID]` should be asserted for that command.
+The SPI_HOST core typically then manages the details of asserting and deasserting the proper `cs_n` line, subject to the timing parameters expressed in {{< regref "CONFIG.CSAAT" >}}, {{< regref "CONFIG.CSNLEAD" >}}, {{< regref "CONFIG.CSNTRAIL" >}}, and {{< regref "CONFIG.CSNIDLE" >}}.
+
+Alternatively the `cs_n` lines can be controlled entirely by firmware.
+If the register {{< regref "CONTROL.MANCS_EN" >}} is set high then the `cs_n` lines are directly controlled by the {{< regref "CONTROL.MANUAL_CS" >}} register.
+
+If [Pass-through mode](#pass-through-mode) is enabled then the `cs_n` lines are controlled by *neither* the SPI_HOST hardware nor the firmware register.
+In Pass-though mode, control of the `cs_n` lines passes directly to the inter-module port, `pt_cs_ni`.
+
+### Arbitrary Length Commands
+
+Typically the length of each commands are limited by the maximum number of bits in the {{< regref "COMMAND.TXN_CNT" >}}, {{< regref "COMMAND.RX_CNT" >}}, and {{< regref "COMMAND.TX1_CNT" >}} fields.
+A pair of SPI_HOST IP commands however can be merged into one long command sequence using the {{< regref "CONFIG.CSAAT" >}} field, which leaves the `cs_n` line low after the first SPI_HOST command.
+By raising the `cs_n` line only after the second command has completed, the peripheral will interpret the resulting transmit/receive sequences as one long command.
+
+This method can be also applied to a series of SPI_HOST IP commands, to create commands of arbitrary length.
+
+### Opportunistic Support for Back-to-back Commands
```

Comment:
do you think this is worth supporting? I wouldn't be against the hardware explicitly blocking updates to any configuration registers while something is ongoing. This part feels a bit weird to me because they can't hit "go" anyways, so I'm not sure supplying a second config / command (a few register writes) will really save that much.

Reply:
How does blocking this work with regtool?  We were under clear instructions from Randall that we should do nothing to stall the TL-UL bus.  @a-pronin what do you think about stalls?

The point here is that the CDC and FSM (at least the way I have them now) are going to have enough state for a command on the backside.   (Maybe I can trim them down?) If there is a command processing then it's certainly not the end of the world to hit go on a second one.

However, there is an overhead to actually verifying this.  So regardless, I can remove this paragraph.

Reply:
the "block" actually works as a regwen mechanism. 
Ie, once a particular "go" bit is triggered, an entire set of configuration registers can no longer be changed until the transaction is finished. 

So when software tries to write to it, it doesn't "stall", it just takes no effect.

Reply:
Hmm.  But then in this case, how does software know that it screwed up?  It still seems like an error to me.

Reply:
Hi Tim,
   I spent some more time thinking about this, and would you mind if we put this down as an issue as well?   There are some subtleties that I still don't understand and I'd like to talk to you more about it.

Reply:
sounds good to me.

Reply:
This comment is to be cited in #5157, and will hopefully be addressed in next PR, #5149.

Reply:
Resolving for now, will take up in issue.